provision/infinispan/create_ispn_clusters.sh

```diff
@@ -0,0 +1,315 @@
+}
+
+function deploy_infinispan_cr_same_cluster() {
+    local kubecfg="${1}"
+    local namespace="${2}"
+    local local_site="${3}"
+    local remote_namespace="${4}"
+    local remote_site="${5}"
+    KUBECONFIG="${kubecfg}" oc apply -f - << EOF
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: infinispan
+  namespace: ${namespace}
+spec:
+  replicas: ${ISPN_REPLICAS}
+  service:
+    type: DataGrid
+    sites:
+      local:
+        name: ${local_site}
+        expose:
+          type: ClusterIP
+        maxRelayNodes: 128
+      locations:
+        - name: ${remote_site}
+          clusterName: infinispan
+          namespace: ${remote_namespace}
+EOF
+}
+
+function deploy_infinispan_cr() {
+    local kubecfg="${1}"
+    local namespace="${2}"
+    local local_site="${3}"
+    local remote_namespace="${4}"
+    local remote_site="${5}"
+    local api_url="${6}"
+    KUBECONFIG="${kubecfg}" oc apply -f - << EOF
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: infinispan
+  namespace: ${namespace}
+spec:
+  replicas: ${ISPN_REPLICAS}
+  service:
+    type: DataGrid
+    sites:
+      local:
+        name: ${local_site}
+        expose:
+          type: Route
+        maxRelayNodes: 128
+        encryption:
+          protocol: TLSv1.3
+          transportKeyStore:
+            secretName: ${XSITE_KS_TLS_SECRET}
+            alias: xsite
+            filename: keystore.p12
+          routerKeyStore:
+            secretName: ${XSITE_KS_TLS_SECRET}
+            alias: xsite
+            filename: keystore.p12
+          trustStore:
+            secretName: ${XSITE_TS_TLS_SECRET}
+            filename: truststore.p12
+      locations:
+        - name: ${remote_site}
+          url: ${api_url}
+          secretName: ${XSITE_TOKEN_SECRET}
+EOF
+}
+
+function deploy_infinispan_cr_without_cross_site() {
+    local kubecfg="${1}"
+    local namespace="${2}"
+    KUBECONFIG="${kubecfg}" oc apply -f - << EOF
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: infinispan
+  namespace: ${namespace}
+spec:
+  replicas: ${ISPN_REPLICAS}
+  service:
+    type: DataGrid
+EOF
+}
+
+function deploy_cache_cr() {
+  local kubecfg="${1}"
+  local namespace="${2}"
+  local cache_name="${3}"
+  local remote_site="${4}"
+  local xsite_mode="${5}"
+  local cr_name=$(echo "${cache_name}" | awk '{print tolower($0)}')
+
+  KUBECONFIG="${kubecfg}" oc apply -f - << EOF
+apiVersion: infinispan.org/v2alpha1
+kind: Cache
+metadata:
+  name: ${cr_name}
+  namespace: ${namespace}
+spec:
+  clusterName: infinispan
+  name: ${cache_name}
+  template: |-
+    distributedCache:
+      mode: "SYNC"
+      owners: "2"
+      statistics: "true"
+      stateTransfer:
+        chunkSize: 16
+      backups:
+        ${remote_site}:
+          backup:
+            strategy: ${xsite_mode}
+            stateTransfer:
+              chunkSize: 16
+EOF
+}
+
+function deploy_cache_cr_without_cross_site() {
```

Comment:
Don't worry, I'm not going to suggest you change it now you've written everything in bash ... but did you consider writing a custom helm chart like the keycloak approach for the CRs? I'm just curious if there were any technical reasons why you couldn't do it that way, or if it was just personal preference.

Reply:
Just because it was quicker. I need to convert the bash to Taskfiles so they can be imported/includes with other Taskfiles. And the helm chart would save some copy-paste too.